File: EIDSS v5/android/EIDSS/MainActivity.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Globalization;
using System.IO;
using System.Threading;
using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;
using eidssdroid.Model;
using eidssdroid.model.Database;


// http://laslow.net/2012/05/10/android-adb-on-windows-x64/

namespace EIDSS
{
    [Activity]
    public class MainActivity : Activity
    {
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetTitle(Resource.String.TitleMainMenu);
            SetContentView(Resource.Layout.MainLayout);

            Button buttonExit = FindViewById<Button>(Resource.Id.ExitButton);
            Button buttonOptions = FindViewById<Button>(Resource.Id.OptionsButton);
            Button buttonHumanCaseList = FindViewById<Button>(Resource.Id.HumanCaseListButton);
            Button buttonNewHumanCase = FindViewById<Button>(Resource.Id.NewHumanCaseButton);

            buttonExit.Click += delegate
            {
                Finish();
            };

            buttonOptions.Click += delegate
            {
                StartActivity(typeof(OptionsActivity));
            };

            buttonHumanCaseList.Click += delegate
            {
                StartActivity(typeof(HumanCaseListActivity));
            };

            buttonNewHumanCase.Click += delegate
            {
                var i = new Intent(ApplicationContext, typeof(HumanCaseActivity));
                var mCase = HumanCase.CreateNew();
                i = i.PutExtra("hc", mCase);
                StartActivityForResult(i, 333);
            };

        }

        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            if (requestCode == 333)
            {
                if (resultCode == Result.Ok)
                {
                    var mCase = (HumanCase)data.GetParcelableExtra("hc");
                    using (var db = new EidssDatabase(this))
                    {
                        db.HumanCaseInsert(mCase);
                    }
                }
            }
            base.OnActivityResult(requestCode, resultCode, data);
        }

    }
}

File: EIDSS v5/android.java/workspace/EIDSS/src/com/bv/eidss/MainActivity.java
package com.bv.eidss;

import com.bv.eidss.data.EidssDatabase;
import com.bv.eidss.model.HumanCase;

import android.os.Bundle;
import android.app.Activity;
import android.content.Intent;
import android.view.View;

//import android.widget.Button;
//import android.view.Menu;

public class MainActivity extends Activity {

	private Activity _this;
	public MainActivity() {
    	_this = this;
	}

	@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle(R.string.TitleMainMenu);
        setContentView(R.layout.activity_main);

    	findViewById(R.id.ExitButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				_this.finish();
			}
		});
    	findViewById(R.id.OptionsButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				_this.startActivity((new Intent()).setClass(getApplicationContext(), OptionsActivity.class));
			}
		});
    	findViewById(R.id.HumanCaseListButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				_this.startActivity((new Intent()).setClass(getApplicationContext(), HumanCaseListActivity.class));
			}
		});
    	findViewById(R.id.NewHumanCaseButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				Intent intent = new Intent();
				intent = intent.setClass(getApplicationContext(), HumanCaseActivity.class);
				intent = intent.putExtra("hc", HumanCase.CreateNew());
				_this.startActivityForResult(intent, 333);
			}
		});
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data)
    {
        if (requestCode == 333)
        {
            if (resultCode == RESULT_OK)
            {
            	HumanCase mCase = (HumanCase)data.getParcelableExtra("hc");
            	EidssDatabase db = new EidssDatabase(this);
                db.HumanCaseInsert(mCase);
                db.close();
            }
        }
        super.onActivityResult(requestCode, resultCode, data);
    }

}
